### ISSUE TITLE
` jackson-databind` should not be optional/provided dependency for Properties, YAML modules

### ISSUE THREAD

**Issue by jvanzyl** [FasterXML/jackson-dataformats-text#184]
In the following pom.xml for jackson-dataformats-properties:

https://github.com/FasterXML/jackson-dataformats-text/blob/master/properties/pom.xml#L23

jackson-databind is not optional, it is required for the JavaPropsMapper to work correctly.

**Comment by petroav** [comment]
To provide some more details on the issue: I have `jackson-dataformat-properties` as a dependency and then attempt to call `new JavaPropsMapper().readValue(file, MyClass.class)` but receive a compiler error:
```
error: cannot access ObjectMapper class file for com.fasterxml.jackson.databind.ObjectMapper not found
```
The `readValue()` method is defined in `ObjectMapper`, which is the super class of `JavaPropsMapper`. `ObjectMapper` is defined in `jackson-databind`. If I add `jackson-databind` as an explicit dependency I get an error that the dependency is unused. I'd rather not turn off the check so for now I've added dummy usage of a class from `jackson-databind`.

**Comment by cowtowncoder** [comment]
Ah yes. For some of dataformats it used to be fully optional (mostly binary codecs), but with 2.10, and especially for Properties, it should be regular full dependency. Will fix for 2.10 and later; check that same is done for CSV and YAML as well -- starting with 2.10, every format has matching `XxxMapper` to support builder pattern so it can not be optional dependency.
